src/main/java/com/example/onlinebookstore/controller/BookController.java

```diff
@@ -25,30 +28,36 @@ public class BookController {
 
     @GetMapping
     @Operation(summary = "Get all books", description = "Get a list of available books")
-    public List<BookDto> getAll(Pageable pageable) {
+    @PreAuthorize("hasRole('ROLE_USER')")
+    public List<BookDto> getAll(@PageableDefault(size = 20, sort = "id",
+            direction = Sort.Direction.ASC) Pageable pageable) {
```

Comment:
I don't sure, but it sort it by Id by default

Reply:
yep, u r right) 
But sometimes,  ID isn`t PK, and the sort uses PK by default
I changed it